### PR TITLE
add AES key wrap support, RFC 3394

### PR DESCRIPTION
**Pull request by cconlon**
This PR adds AES key wrap support, along with test vectors from the RFC.

The user can enable by using "--enable-aeskeywrap", or by defining HAVE_AES_KEYWRAP.

### CODE REVIEW

File: wolfcrypt/src/aes.c [hunk 1]
```diff
@@ -4504,6 +4504,179 @@ int  wc_AesCcmDecrypt(Aes* aes, byte* out, const byte* in, word32 inSz,
...
+            return;
+    }
+}
+
+/* Decrement key wrap counter */
+static INLINE void DecrementKeyWrapCounter(byte* inOutCtr)
+{
+    int i;
+
+    for (i = 0; i <= KEYWRAP_BLOCK_SIZE - 1; i++) {
```
Comment (toddouska) [inline wolfcrypt/src/aes.c]: Should be same for loop as Increment

File: wolfcrypt/src/aes.c [hunk 2]
```diff
@@ -4504,6 +4504,179 @@ int  wc_AesCcmDecrypt(Aes* aes, byte* out, const byte* in, word32 inSz,
...
+        if (inOutCtr[i])
```
Comment (toddouska) [inline wolfcrypt/src/aes.c]: we're done unless we underflow

if (--inOutCtr[i] == 0xff)
    return;

Comment (cconlon) [inline wolfcrypt/src/aes.c]: Thanks!  Making these changes and will re-push shortly.

File: wolfcrypt/src/aes.c [hunk 3]
```diff
@@ -4504,6 +4504,179 @@ int  wc_AesCcmDecrypt(Aes* aes, byte* out, const byte* in, word32 inSz,
...
+    }
+}
+
+/* Decrement key wrap counter */
+static INLINE void DecrementKeyWrapCounter(byte* inOutCtr)
+{
+    int i;
+
+    for (i = KEYWRAP_BLOCK_SIZE - 1; i >= 0; i--) {
+        if (inOutCtr[i])
```
Comment (toddouska) [inline wolfcrypt/src/aes.c]: This if check is wrong, think about 256.

**Comment by cconlon** [comment]
Fixes made, ready for re-review.

File: wolfcrypt/src/aes.c [hunk 4]
```diff
@@ -4504,6 +4504,179 @@ int  wc_AesCcmDecrypt(Aes* aes, byte* out, const byte* in, word32 inSz,
...
+            if (--inOutCtr[i] & 0xff)
```
Comment (toddouska) [inline wolfcrypt/src/aes.c]: We're not done on underflow.  We're done unless underflow.

if (--inOutCtr[i] != 0xff)
return;

**Pull request merged**